daemon/src/surface.rs

```diff
@@ -188,26 +207,31 @@ impl Surface {
                 }
             }
             loop {
-                let files: Vec<PathBuf> = WalkDir::new(path)
-                    .into_iter()
-                    .filter_map(|e| e.ok())
-                    .filter(|e| {
-                        if let Some(guess) = new_mime_guess::from_path(e.path()).first() {
-                            guess.type_() == "image"
-                        } else {
-                            false
+                let is_zero = self.current_index == 0;
+                let is_below_len =
+                    !self.image_paths.is_empty() && self.current_index < self.image_paths.len();
```

Comment:
One thing that I am not sure about is: what happens when the timer expires and a new image should be drawn? The current_index would point to a valid image and it draw the current one. Maybe we can increase `current_index` if the timer expired so that a new image would be drawn.

Reply:
That's a very good point, didn't think of that. I'll have a look at how to fix this